### PR TITLE
internal/gitserver: Remove unused clientImplementor.ConnForRepo

### PR DESCRIPTION
**Pull request by indradhanush**
## Test plan

- CI 
- Local dev instance works

### CODE REVIEW

**Review by sashaostrikov — APPROVED** [review]

**Review by mrnugget — APPROVED** [review]

File: internal/gitserver/client.go [hunk 1]
```diff
@@ -483,11 +483,6 @@ func (c *clientImplementor) ClientForRepo(ctx context.Context, repo api.RepoName
 	return c.clientSource.ClientForRepo(ctx, c.userAgent, repo)
 }
 
-// TODO: Looks like this is not used anywhere.
-func (c *clientImplementor) ConnForRepo(ctx context.Context, repo api.RepoName) (*grpc.ClientConn, error) {
-	return c.clientSource.ConnForRepo(ctx, c.userAgent, repo)
```
Comment (sashaostrikov) [inline internal/gitserver/client.go:488]: What about this function of a client?

Comment (sashaostrikov) [reply internal/gitserver/client.go:488]: Can `c.clientSource.ConnForRepo` become unused with the removal of an unused function?

Comment (indradhanush) [reply internal/gitserver/client.go:488]: Yeah, good point. It looks like it can be removed as well. I will follow up in another PR.

Comment (indradhanush) [reply internal/gitserver/client.go:488]: Follow up is here: https://github.com/sourcegraph/sourcegraph/pull/55419

**Pull request merged**